src/Illuminate/Http/Client/Pool.php

```diff
@@ -65,7 +74,7 @@ public function as(string $key)
      */
     protected function asyncRequest()
     {
-        return $this->factory->setHandler($this->handler)->async();
+        return (clone $this->pendingRequest)->async();
```

Comment:
Is there a specific reason to clone the request instead of just re-using the existing `PendingRequest`? In the end it are the HTTP request methods like get, post, patch, etc. that are responsible for creating the promises when the `PendingRequest` has an `async` property which is set to `true`.

Reply:
In PHP, objects are passed by reference, so as soon as you modify a Pool instance (via `as()` for instance), it will also be modifying the underlying `PendingRequest`. This means calling `Http::pool(fn($pool) => [$pool->as('item1')->get('google.com'), $pool->as('item2')->get('laravel.com')])` is relying on the same PendingRequest instance unless we clone it.